### PR TITLE
feat: Add convenience methods for annotation lifecycle

### PR DESCRIPTION
**Pull request by asgeirrr**

### CODE REVIEW

File: rossum_api/elis_api_client_sync.py [hunk 1]
```diff
@@ -282,6 +302,16 @@ def update_part_annotation(self, annotation_id: int, data: Dict[str, Any]) -> An
             self.elis_api_client.update_part_annotation(annotation_id, data)
         )
 
+    def update_annotation_data(self, annotation_id: int, operations: List[Dict[str, Any]]) -> None:
+        """https://elis.rossum.ai/api/docs/#update-annotation-data"""
+        self.event_loop.run_until_complete(
+            self.elis_api_client.update_annotation_data(annotation_id, operations)
+        )
```
Comment (bara-m) [inline rossum_api/elis_api_client_sync.py:309]:
```suggestion
    def bulk_update_annotation_data(self, annotation_id: int, operations: List[Dict[str, Any]]) -> None:
        """https://elis.rossum.ai/api/docs/#bulk-update-annotation-data"""
        self.event_loop.run_until_complete(
            self.elis_api_client.bulk_update_annotation_data(annotation_id, operations)
        )
```

File: tests/elis_api_client/test_annotations.py [hunk 2]
```diff
@@ -390,6 +461,57 @@ def is_imported(annotation):
...
+        with patch("asyncio.sleep") as sleep_mock:
+            annotation = client.upload_and_wait_until_imported(
+                queue_id=8199,
+                filepath="tests/data/sample_invoice.pdf",
+                filename="document.pdf",
+                sleep_s=2,
+                sideloads=["content"],
+            )
+
+        assert annotation == Annotation(**dummy_annotation)
```
Comment (bara-m) [inline tests/elis_api_client/test_annotations.py:502]: Shouldn't this be

```python
assert annotation == Annotation(**{**dummy_annotation, "content": []})
```

as in the test above?

File: tests/elis_api_client/test_annotations.py [hunk 3]
```diff
@@ -239,6 +239,56 @@ def is_imported(annotation):
...
+        http_client.fetch_one.side_effect = [in_progress_annotation, dummy_annotation]
+        # Return sideloaded content
+        http_client.request_json.return_value = {"content": []}
+
+        with patch("asyncio.sleep") as sleep_mock:
+            annotation = await client.poll_annotation_until_imported(
+                dummy_annotation["id"], sleep_s=2, sideloads=["content"]
+            )
+
+        assert annotation == Annotation(**dummy_annotation)
```
Comment (bara-m) [inline tests/elis_api_client/test_annotations.py:255]: Shouldn't this be

```python
assert annotation == Annotation(**{**dummy_annotation, "content": []})
```

as in the test of the sync client?

File: tests/elis_api_client/test_annotations.py [hunk 4]
```diff
@@ -239,6 +239,56 @@ def is_imported(annotation):
...
+        sleep_mock.assert_called_once_with(2)
+
+    async def test_start_annotation(self, elis_client, dummy_annotation):
+        client, http_client = elis_client
+        http_client.replace.return_value = dummy_annotation
+
+        aid = dummy_annotation["id"]
+
+        await client.start_annotation(aid)
+        http_client.request_json.assert_called_with("POST", "annotations/314528/start")
```
Comment (bara-m) [inline tests/elis_api_client/test_annotations.py:290]:
```suggestion
        http_client.request_json.assert_called_with("POST", f"annotations/{aid}/start")
```

I would prefer this to make it clear where is the number coming from. WDYT?

**Review by bara-m — CHANGES_REQUESTED** [review]

File: tests/elis_api_client/test_annotations.py (hunk 4 above)
Comment (asgeirrr) [reply tests/elis_api_client/test_annotations.py:290]: I usually prefer to hardcode the values in asserts but why not in this case...

File: tests/elis_api_client/test_annotations.py (hunk 3 above)
Comment (asgeirrr) [reply tests/elis_api_client/test_annotations.py:255]: Ha, you've uncovered a side-effect in the tests that caused you didn't have to add `content` to the expected annotation, I've fixed it everywhere :wink:

File: rossum_api/elis_api_client.py [hunk 5]
```diff
@@ -302,6 +302,27 @@ async def poll_annotation(
...
+        )
+
+    async def upload_and_wait_until_imported(
+        self, queue_id: int, filepath: Union[str, pathlib.Path], filename: str, **poll_kwargs
+    ) -> Annotation:
+        """A shortcut for uploading a single file and waiting until its annotation is imported."""
+        (annotation_id,) = await self.import_document(queue_id, [(filepath, filename)])
+        return await self.poll_annotation_until_imported(annotation_id, **poll_kwargs)
+
+    async def start_annotation(self, annotation_id: int) -> None:
```
[the remark this answers is not in the document]
Comment (asgeirrr) [reply rossum_api/elis_api_client.py:320]: It returns the content of the annotation and I would need to add a lot more models to `rossum-api` so I didn't went for it until someone needs it.

[the remark this answers is not in the document]
Comment (bara-m) [reply rossum_api/elis_api_client.py:320]: Ok. We are using the `/start` endpoint in some projects but I am not sure if we need the response. Let's add it later if we will need it.

File: tests/elis_api_client/test_annotations.py [hunk 6]
```diff
@@ -269,6 +323,27 @@ async def test_update_part_annotation(self, elis_client, dummy_annotation):
...
+    async def test_bulk_update_annotation_data(self, elis_client, dummy_annotation):
+        client, http_client = elis_client
+        http_client.request_json.return_value = dummy_annotation
+
+        aid = dummy_annotation["id"]
+        operations = [{"id": 2510559656, "op": "remove"}, {"id": 2510559657, "op": "remove"}]
+        await client.bulk_update_annotation_data(aid, operations)
+
+        http_client.request_json.assert_called_with(
+            "POST", f"annotations/{314528}/content/operations", json={"operations": operations}
```
Comment (bara-m) [inline tests/elis_api_client/test_annotations.py:335]:
```suggestion
            "POST", f"annotations/{aid}/content/operations", json={"operations": operations}
```

**Review by bara-m — CHANGES_REQUESTED** [review]

**Review by bara-m — APPROVED** [review]

**Pull request merged**